src/jrd/ExtEngineManager.cpp

```diff
@@ -703,6 +717,13 @@ void* ExtEngineManager::ExternalContextImpl::setInfo(int code, void* value)
 	return oldValue;
 }
 
+void ExtEngineManager::ExternalContextImpl::checkExternalAttachment()
+{
+	if (!externalAttachment)
+	{
+		(Arg::Gds(isc_random) << Arg::Str("Use of system attachment in UDR is not allowed")).raise();
```

Comment:
We need special error code here so UDR may know why it was not possible to get the attachment.

Reply:
Sure